src/data/chats/chat_entity.rs

```diff
@@ -0,0 +1,38 @@
+//! Hopefully, provisional solution to unify model files and agents in the chat system.
+
+use moly_mofa::MofaAgent;
+use moly_protocol::data::{File, FileID};
+use serde::{Deserialize, Serialize};
+
+/// Identifies either a model file or an agent.
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub enum ChatEntityId {
+    ModelFile(FileID),
+    /// Since agents are currently fixed enum values, the agent itself is the identifier.
+    Agent(MofaAgent),
+}
+
+/// Reference to either a model file or an agent.
+///
+/// Can be used to chain iterators of both types or simply to take either as a parameter.
+#[derive(Debug, Clone, Serialize, Copy)]
+pub enum ChatEntityRef<'a> {
```

Comment:
After some cleanup this is only used at `prompt_input.rs` and `entity_button.rs`.